pandas_dev_flaker/_ast_helpers.py

```diff
@@ -37,13 +37,40 @@ def is_str_constant(
     node: ast.Call,
 ) -> bool:
     return isinstance(node.func, ast.Attribute) and (
-        (
-            sys.version_info < (3, 8)
-            and isinstance(node.func.value, ast.Str)
-        )
+        (sys.version_info < (3, 8) and isinstance(node.func.value, ast.Str))
         or (
             sys.version_info >= (3, 8)
             and isinstance(node.func.value, ast.Constant)
             and isinstance(node.func.value.value, str)
         )
     )
+
+
+def parse_annotation(node: ast.BinOp) -> bool:
+    series, any_array_like = "Series", "AnyArrayLike"
+    is_series, is_array_like = False, False
+
+    while isinstance(node, ast.BinOp):
+        if isinstance(node.right, ast.Name):
+            if node.right.id == series:
+                is_series = True
+            if node.right.id == any_array_like:
+                is_array_like = True
+        else:
+            raise AssertionError
+
+        if isinstance(node.left, ast.Name):
+            if node.left.id == series:
+                is_series = True
+            if node.left.id == any_array_like:  # pragma: no cover
+                is_array_like = True
```

Comment:
why can't this branch be covered?

Reply:
on line 62, if the `if` condition is true, we're to break out of the while loop. but before breaking out we're also making checks for series and any-array-like annotations, these 2 inner conditions were affecting the test coverage.

the coverage warning: `line 65 didn't jump to line 67 because the condition for line 65 was never false`

i wrote tests that covered for when line 65 was false. i'm sincerely not really sure why it fails. 
both conditions have to be checked because either of them could be the first annotation

Reply:
sure, thanks for explaining - I'll take a look later then

Reply:
this doesn't pass